roles/setup_repo/tasks/setup_repo.yml

```diff
@@ -27,8 +28,20 @@
 
 - name: Validate Credentials
   fail:
-      msg: "repo_username = {{ repo_username }} or repo_password = {{ repo_password }} are not valid!."
-  when: enable_edb_repo|bool and (repo_username | length < 1 or repo_password | length < 1)
+      msg: "repo_username = {{ repo_username }} or repo_password = {{ repo_password }} or repo_token = {{ repo_token }} are not valid!."
+  when:
+    - enable_edb_repo|bool
+    - repo_token | length < 1 and (repo_username | length < 1 or repo_password | length < 1)
+
+- name: Define EDB repo script link for RedHat family
+  set_fact:
+      edb_repo_script_link: "https://downloads.enterprisedb.com/{{ repo_token }}/enterprise/setup.rpm.sh"
```

Comment:
Wondering if this should be defined along with other variables in `vars/`

Reply:
Instead of `vars`, I used if condition for `edb_repo_script_link` in `defaults/main.yml`. 
The main reason for not using `vars` is that if we use it, we also need to focus on refactoring and reusing other variables based on `ansible_os_family`, which is a good amount of work, and we will pick up this work later and submit it as a different PR.